### ISSUE TITLE
Credentials to access COP DEM on AWS?

### ISSUE THREAD

**Issue by fnattino** [CityRiverSpaces/CRiSp#46]
Are credentials needed in order to access the DEM tile? The STAC catalog includes  `s3://` links as assets, not sure whether terra (GDAL) can access these without credentials, even if data is in a free bucket

**Comment by cforgaci** [comment]
Yes, I needed to set up credentials to be able to get the DEM. I also received a free tier usage limit alert from AWS. I am not sure, maybe we should use `FILE` as default and `STAC` as alternative, also informing the user that they need to set up credentials.

**Comment by fnattino** [comment]
Some more details on what to check here. 
 
In `valley.R`, `get_dem()` is the function that retrieves the DEM. The following snippet is supposed to access the DEM for a given region:

```r
# devtools::load_all()
bb <- get_osm_bb("Bucharest")
dem <- get_dem(bb)
```

Within `get_dem` > `load_raster`, `terra::rast(url)` is the function that actually loads the raster from the URL/path (it is based on GDAL).

- Can we run the snippet above without setting credentials? This would work if the `s3://...` URLs returned by the STAC API (queried in `get_stac_asset_urls`) can be accessed without authentication.
- If not, can you convert the `s3://...` into a `https://` URL that can be accessed without authentication? 
- If not, how can you set up credentials so that one it works (i.e. how can you set up authentication so that you can access the data)? Description should be added to a docstring in `valley.R`.

**Comment by fnattino** [comment]
@vanlankveldthijs as discussed earlier please use the branch `52-corridor-valley-fn` until #59 is merged

**Comment by fnattino** [comment]
> [@vanlankveldthijs](https://github.com/vanlankveldthijs) as discussed earlier please use the branch `52-corridor-valley-fn` until [#59](https://github.com/CityRiverSpaces/CRiSp/pull/59) is merged

@vanlankveldthijs  #59 is merged, feel free to use the main branch now!

**Comment by SarahAlidoost** [comment]
I copy-paste the error here when running the readme example with main branch:
```R
> bucharest_river <- delineate_corridor("Bucharest", "Dâmbovița", crs = epsg_code)
Error: [rast] file does not exist: /vsis3/copernicus-dem-30m/Copernicus_DSM_COG_10_N44_00_E026_00_DEM/Copernicus_DSM_COG_10_N44_00_E026_00_DEM.tif
In addition: Warning message:
In new_CppObject_xp(fields$.module, fields$.pointer, ...) :
  GDAL Error 15: AWS_SECRET_ACCESS_KEY and AWS_NO_SIGN_REQUEST configuration options not defined, and ~/.aws/credentials not filled
```

**Comment by fnattino** [comment]
Thanks for reporting the details of the error @SarahAlidoost! Indeed, I have the file `~/.aws/credentials` in place. By googling a bit it seems that by setting the environment variable `AWS_NO_SIGN_REQUEST=YES` one should be able to access public buckets without authentication. Could you maybe try to set in R (with `Sys.setenv()`?) and see whether you can retrieve DEM data? 

If this does not work, I guess one would need to create a AWS account and setup the `~/.aws/credentials` file. We should provide instructions on how to do that though (or at least link to such instructions).

**Comment by SarahAlidoost** [comment]
> Thanks for reporting the details of the error [@SarahAlidoost](https://github.com/SarahAlidoost)! Indeed, I have the file `~/.aws/credentials` in place. By googling a bit it seems that by setting the environment variable `AWS_NO_SIGN_REQUEST=YES` one should be able to access public buckets without authentication. Could you maybe try to set in R (with `Sys.setenv()`?) and see whether you can retrieve DEM data?
> 
> If this does not work, I guess one would need to create a AWS account and setup the `~/.aws/credentials` file. We should provide instructions on how to do that though (or at least link to such instructions).

nice job on finding the solution 👍  setting env variable fixes the issue, see #87
